src/libsystemd/sd-journal/journal-vacuum.c

```diff
@@ -159,6 +159,9 @@ int journal_directory_vacuum(
                 if (!S_ISREG(st.st_mode))
                         continue;
 
+                size = 512UL * (uint64_t) st.st_blocks;
+                sum += size;
```

Comment:
this isn't right. we should still ignore all non-journal files, that have names that we don't recognize at all. and empty files should not be accounted either as we already delete them here, and hence definitely don't matter anymore

Reply:
i.e. we should count everything that we do n_active_files++ for in the branches below.